### PR TITLE
WIP: Add metadata to the JSON payload and acceptable params

### PR DESCRIPTION
**Pull request by gabelimon**
@themartorana I started a PR based on my issue (https://github.com/themartorana/python-postmark/issues/74). Is this about right? I'll start adding tests next time I look at this.

### CODE REVIEW

**Comment by themartorana** [comment]
LGTM. @nicholasserra is the admin on this repo now so I'll let him have a look and merge. Thanks for the contribution!

**Comment by gabelimon** [comment]
@nicholasserra It's a WIP only because I don't know the testing procedures of this repo. If this PR looks good to go already I can just manually test it and take off the WIP label.

**Comment by nicholasserra** [comment]
Okay. Please do manually test. There's also a test.py where unit tests live, should add some to that file that cover this case. I'll run through it when I get a chance and see if theres any more test cases to add.

**Comment by nicholasserra** [comment]
Just manually tested this. Seems to be working fine with and without metadata. I checked my sent emails in my postmark account and metadata was set correctly. I have a nitpick comment in the code i'm about to add.

Let me know if you're okay with adding some tests, or I can add some. Thanks!

File: postmark/core.py [hunk 1]
```diff
@@ -90,6 +90,7 @@ def __init__(self, **kwargs):
         self.__attachments = []
         self.__message_id = None
         #self.__multipart = False
+        self.__metadata = {}
         self.__template_id = None
```
Comment (nicholasserra) [inline postmark/core.py]: Can you also add metadata to the docstring on this method?

File: postmark/core.py [hunk 2]
```diff
@@ -151,6 +153,18 @@ def _set_custom_headers(self, value):
...
+        '''
+        A special set function to ensure
+        we're setting with a dictionary
+        '''
+        if value is None:
+            setattr(self, '_PMMail__metadata', {})
+        elif isinstance(value, dict):
+            setattr(self, '_PMMail__metadata', value)
+        else:
+            raise TypeError('Metadata must be a dictionary of key-value pairs')
```
Comment (nicholasserra) [inline postmark/core.py]: Can you also check that the metadata dict is a simple `string: string` dictionary? I just tried sending with a more complex set of nested dictionaries and got back an error. We should handle that in here, in my opinion. Thanks!

**Comment by gabelimon** [comment]
Hey @nicholasserra, we didn't end up implementing the feature that we were going to use this for so that's why I haven't looked at this in a bit. Let me see if I can find time today to finish this up since your feedback is pretty straightforward.

**Comment by nicholasserra** [comment]
Awesome sounds good, no big deal if you don't have time, I can get it across the finish line also

**Comment by nicholasserra** [comment]
Fixed this up and merged, thanks! I found some larger issues with type checking in the package, so i'll fix that and then add some tests for this later.

**Pull request merged**